pom.xml

```diff
@@ -56,8 +58,109 @@
+                            </execution>
+                            <execution>
+                                <id>docker-generate-binaries</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <arguments>
+                                        <argument>run</argument>
+                                        <argument>-i</argument>
+                                        <argument>--rm</argument>
+                                        <argument>--mount</argument>
+                                        <argument>type=bind,src=${basedir}/${binary.src},dst=/org/jenkinsci/plugins/durabletask</argument>
+                                        <argument>${binary.dockerimage}:${revision}</argument>
+                                    </arguments>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>docker-image-cleanup</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <arguments>
+                                        <argument>rmi</argument>
+                                        <argument>${binary.dockerimage}:${revision}</argument>
+                                    </arguments>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-resources-plugin</artifactId>
+                        <version>3.1.0</version>
+                        <executions>
+                            <execution>
+                                <id>copy-binaries</id>
+                                <phase>process-resources</phase>
+                                <goals>
+                                    <goal>copy-resources</goal>
+                                </goals>
+                                <configuration>
+                                    <outputDirectory>src/main/resources/org/jenkinsci/plugins/durabletask</outputDirectory>
+                                    <resources>
+                                        <resource>
+                                            <directory>src/main/go/org/jenkinsci/plugins/durabletask</directory>
+                                            <includes>
+                                                <include>heartbeat-launcher-*</include>
```

Comment:
This should not have been built inside the `src/` tree to begin with. Have the external script copy what it needs to `target/` and do all further processing there.

Reply:
will include that in the script i am adding